gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeCatalogTest.java

```diff
@@ -0,0 +1,346 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.bigquery.biglake.v1.Catalog;
+import com.google.cloud.bigquery.biglake.v1.CatalogName;
+import com.google.cloud.bigquery.biglake.v1.Database;
+import com.google.cloud.bigquery.biglake.v1.DatabaseName;
+import com.google.cloud.bigquery.biglake.v1.HiveDatabaseOptions;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.CatalogTests;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.junit.Rule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class BigLakeCatalogTest extends CatalogTests<BigLakeCatalog> {
+
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+  @TempDir public Path temp;
+
+  private static final String GCP_PROJECT = "my-project";
+  private static final String GCP_REGION = "us";
+  private static final String CATALOG_ID = "biglake";
+
+  private String warehouseLocation;
+
+  private BigLakeCatalog fakeBigLakeCatalog;
+
+  private BigLakeClient mockBigLakeClient;
+  private BigLakeCatalog mockBigLakeCatalog;
+
+  @BeforeEach
+  public void createCatalog() throws Exception {
+    File warehouse = temp.toFile();
+    warehouseLocation = warehouse.getAbsolutePath();
+
+    ImmutableMap<String, String> properties =
+        ImmutableMap.of(
+            BigLakeCatalog.PROPERTIES_KEY_GCP_PROJECT,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation);
+
+    BigLakeClient fakeBigLakeClient = new FakeBigLakeClient();
+    fakeBigLakeCatalog = new BigLakeCatalog();
+    fakeBigLakeCatalog.setConf(new Configuration());
+    fakeBigLakeCatalog.initialize(
+        CATALOG_ID, properties, GCP_PROJECT, GCP_REGION, fakeBigLakeClient);
+
+    mockBigLakeClient = mock(BigLakeClient.class);
+    mockBigLakeCatalog = new BigLakeCatalog();
+    mockBigLakeCatalog.setConf(new Configuration());
+    mockBigLakeCatalog.initialize(
+        CATALOG_ID, properties, GCP_PROJECT, GCP_REGION, mockBigLakeClient);
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  @Override
+  protected BigLakeCatalog catalog() {
+    return fakeBigLakeCatalog;
+  }
+
+  // By pass this test from CatalogTests, because BigLake API does not support "/" in resource IDs.
+  @Test
+  public void testNamespaceWithSlash() {}
```

Comment:
Can you replace this with a test of the actual behavior? I'm assuming that you get a nice error message.

Reply:
Done. Added the tests that validation of resource IDs fails.